gemma-core/src/main/java/ubic/gemma/persistence/service/expression/arrayDesign/ArrayDesignDaoImpl.java

```diff
@@ -595,10 +605,16 @@ protected void postProcessValueObjects( List<ArrayDesignValueObject> results ) {
     @Override
     public List<ArrayDesignValueObject> loadValueObjectsForEE( @Nullable Long eeId ) {
         if ( eeId == null ) {
-            return new ArrayList<>();
+            return Collections.emptyList();
         }
-        Collection<Long> ids = CommonQueries.getArrayDesignIdsUsed( eeId,
-                this.getSessionFactory().getCurrentSession() );
+        //noinspection unchecked
+        List<Long> ids = this.getSessionFactory().getCurrentSession()
+                .createQuery( "select distinct ad.id from ExpressionExperiment as ee "
+                        + "join ee.bioAssays b join b.arrayDesignUsed ad "
+                        + "where ee.id = :eeId" )
+                .setParameter( "eeId", eeId )
+                .setCacheable( true )
+                .list();
```

Comment:
This could be a method in ArrayDesignDao, but maybe it would be best to reuse the one that retrieve entities instead.